src/openApiRequestValidator.js

```diff
@@ -0,0 +1,301 @@
+                throw ValidationError(message, 400);
+            }
+
+            if (
+                parameter.schema &&
+                parameter.schema.type === 'array' &&
+                !parameter.explode
+            ) {
+                const delimiter = arrayDelimiter[parameter.style];
+                if (!delimiter) {
+                    const message = `Parameter 'style' has incorrect value '${parameter.style}' for [${parameter.name}]`;
+                    throw ValidationError(message, 400);
+                }
+                parseArray.push({
+                    name,
+                    reqField,
+                    delimiter
+                });
+            }
+
+            if (
+                parameter.schema &&
+                parameter.schema.type === 'array' &&
+                parameter.explode
+            ) {
+                parseArrayExplode.push({
+                    name,
+                    reqField
+                });
+            }
+
+            if (!schema[reqField].properties) {
+                schema[reqField] = {
+                    type: 'object',
+                    properties: {},
+                };
+            }
+
+            schema[reqField].properties[name] = parameterSchema;
+            if (parameter.required) {
+                if (!schema[reqField].required) {
+                    schema[reqField].required = [];
+                }
+                schema[reqField].required.push(name);
+            }
+        });
+
+        return {
+            schema,
+            parseJson,
+            parseArray,
+            parseArrayExplode
+        };
+    }
+
+    requestBodyToSchema (path, contentType, requestBody) {
+        if (requestBody.content) {
+            const content = requestBody.content[type];
+
+            if (!content) {
+                const message = !contentType == null
+                    ? 'media type not specified'
+                    : `unsupported media type ${contentType.contentType}`;
+                throw ValidationError(message, 415);
+            }
+
+            const schema = this.cleanseContentSchema(contentType, requestBody);
+            return schema || content.schema || {};
+        }
+        return {};
+    }
+
+    cleanseContentSchema (contentType, requestBody) {
+        const bodyContentSchema =
+          requestBody.content[contentType] &&
+          requestBody.content[contentType].schema;
+    
+        let bodyContentRefSchema = null;
+        if (bodyContentSchema && '$ref' in bodyContentSchema) {
+          const objectSchema = this.ajv.getSchema(bodyContentSchema.$ref);
+          bodyContentRefSchema =
+            objectSchema &&
+            objectSchema.schema &&
+            objectSchema.schema.properties
+              ? { ...objectSchema.schema }
+              : null;
+        }
+        // handle readonly / required request body refs
+        // don't need to copy schema if validator gets its own copy of the api spec
```

Comment:
In this case, is sharing the same spec desirable for consistency? Why would validator want/need its own copy, how would that look?

Reply:
The validator should have a copy of of the full schema when initialized so that we can use `ajv` to get the desired schema at anytime. But I think this comment has to do with getting the schema by filtering the original spec instead of copying the whole thing.